libshaderc_spvc/include/shaderc/spvc.h

```diff
@@ -0,0 +1,149 @@
+            #define SHADERC_EXPORT __attribute__((visibility("default")))
+        #else
+            #define SHADERC_EXPORT
+        #endif
+    #endif
+#else
+    #define SHADERC_EXPORT
+#endif
+
+//XXX share with libshaderc?
+// Indicate the status of a compilation.
+typedef enum {
+  spvc_compilation_status_success = 0,
+  spvc_compilation_status_invalid_stage,  // error stage deduction
+  spvc_compilation_status_compilation_error,
+  spvc_compilation_status_internal_error,  // unexpected failure
+  spvc_compilation_status_null_result_object,
+  spvc_compilation_status_invalid_assembly,
+} spvc_compilation_status;
+
+// An opaque handle to an object that manages all compiler state.
+typedef struct spvc_compiler* spvc_compiler_t;
+
+// Returns a spvc_compiler_t that can be used to compile modules.
+// A return of NULL indicates that there was an error initializing the compiler.
+// Any function operating on spvc_compiler_t must offer the basic
+// thread-safety guarantee.
+// [http://herbsutter.com/2014/01/13/gotw-95-solution-thread-safety-and-synchronization/]
+// That is: concurrent invocation of these functions on DIFFERENT objects needs
+// no synchronization; concurrent invocation of these functions on the SAME
+// object requires synchronization IF AND ONLY IF some of them take a non-const
+// argument.
+SHADERC_EXPORT spvc_compiler_t spvc_compiler_initialize(void);
+
+// Releases the resources held by the spvc_compiler_t.
+// After this call it is invalid to make any future calls to functions
+// involving this spvc_compiler_t.
+SHADERC_EXPORT void spvc_compiler_release(spvc_compiler_t);
+
+// An opaque handle to an object that manages options to a single compilation
+// result.
+typedef struct spvc_compile_options* spvc_compile_options_t;
+
+// Returns a default-initialized spvc_compile_options_t that can be used
+// to modify the functionality of a compiled module.
+// A return of NULL indicates that there was an error initializing the options.
+// Any function operating on spvc_compile_options_t must offer the
+// basic thread-safety guarantee.
+SHADERC_EXPORT spvc_compile_options_t
+    spvc_compile_options_initialize(void);
+
+// Returns a copy of the given spvc_compile_options_t.
+// If NULL is passed as the parameter the call is the same as
+// spvc_compile_options_init.
+SHADERC_EXPORT spvc_compile_options_t spvc_compile_options_clone(
+    const spvc_compile_options_t options);
+
+// Releases the compilation options. It is invalid to use the given
+// spvc_compile_options_t object in any future calls. It is safe to pass
+// NULL to this function, and doing such will have no effect.
+SHADERC_EXPORT void spvc_compile_options_release(
+    spvc_compile_options_t options);
+
+// An opaque handle to the results of a call to any spvc_compile_into_*()
+// function.
+typedef struct spvc_compilation_result* spvc_compilation_result_t;
+
+// Takes SPIR-V as a sequence of 32-bit words and compiles to GLSL.
```

Comment:
Say that the SPIR-V must pass validation.
Which then begs the question of how to specify validation options.

One choice is to set, as an additional option, an spv_validator_options object as defined by spirv-tools/libspirv.h
(and then also include that header at the top of this file too.)